src/Uno.UI/UI/Xaml/Controls/Control/Control.cs

```diff
@@ -173,22 +173,28 @@ private protected virtual object CoerceIsEnabled(object baseValue, DependencyPro
 				return false;
 			}
 
-			// The baseValue hasn't been set yet, so we need to make sure we're not
+			// The baseValue hasn't been set inside PropertyDetails yet, so we need to make sure we're not
 			// reading soon-to-be-outdated values
-			var localValue = precedence < DependencyPropertyValuePrecedences.Inheritance ? // < actually means higher precedence
-				baseValue :
-				((IDependencyObjectStoreProvider)this).Store.GetPropertyDetails(IsEnabledProperty).GetValueUnderPrecedence(DependencyPropertyValuePrecedences.Coercion).value;
-			var parentValue = precedence == DependencyPropertyValuePrecedences.Inheritance ?
-				baseValue :
-				((IDependencyObjectStoreProvider)this).Store.GetPropertyDetails(IsEnabledProperty).GetValue(DependencyPropertyValuePrecedences.Inheritance);
+
+			var parentValue = baseValue;
+			DependencyPropertyDetails propertyDetails = null;
+
+			if (precedence != DependencyPropertyValuePrecedences.Inheritance)
+			{
+				propertyDetails = ((IDependencyObjectStoreProvider)this).Store.GetPropertyDetails(IsEnabledProperty);
+				parentValue = propertyDetails.GetValue(DependencyPropertyValuePrecedences.Inheritance);
```

Comment:
You should be able to use GetValue with the specific precedence parameter to get the value at the precedence that you need. It's best to avoid using property details outside DependencyObjectStore.

Reply:
GetValue internally gets the property details anyway. I'm just shortcutting having to go through multiple frames just to get the property details again later. I don't mind it either way, so just confirm again if you want me to swap out the `GetPropertyDetails` calls.

Reply:
I think the extra frames aren't an issue.

Reply:
Indeed, the extra frames are not an issue. They may get inlined as needed, but it's generally not a concern.